dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
> There has been some discussion about the possibility of MeshShape being changed into something that only holds geometric data. If we do that, then we could have it so that the texture and color data gets automatically put into the VisualAddon as it's parsed with assimp. I think this would be also be nice for moving away from assimp being a core dependency (instead it would only be an io dependency).

:+1: for moving away from Assimp in the core API. In theory, I like the idea of texture and color data getting stored inside `VisualAddon`. This would work okay for vertex color, but I am concerned about the feasibility of doing this with textures. Assimp has a dozen different types of textures and material properties. My impression is that it's the wild west, especially for meshes loaded from COLLADA.

@mklingen has much more experience here. Hopefully he can provide some illumination about what is actually used.


> It would probably be best to tackle alongside the new GLFW-based rendering library that has been discussed.

What would be the relationship between `SimWindow`, `osgDart`, and the new GLFW-based rendering library? I also hope this allows us to remove the `draw` functions from the `dynamics` API - I really don't think rendering logic belongs there.

> I don't know if this would be silly or not, but we could potentially make `VisualAddon` be an `AddonManager` and be able to attach a `MeshColorAddon` which could contain per-vertex color information.

I am against this for a few reasons:

1. It adds a lot of complexity to an already-complex part of DART.
2. What is the advantage of putting per-vertex colors in separate `MeshColorAddOn` versus directly inside `VisualAddOn`?
3. `MeshColorAddon` only makes sense on `MeshShape`s, but may be added to any type of `ShapeNode`.